components/Form/MultiStepForm.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Formik } from 'formik';
+import { getErrorMessage } from 'common/utils/api-utils';
+import { capitalizeFirstLetter } from 'common/utils/string-utils';
+import Button from 'components/Button/Button';
+import Form from 'components/Form/Form';
+import Alert from 'components/Alert/Alert';
+import styles from './MultiStepForm.css';
+
+class MultiStepForm extends React.Component {
+  static propTypes = {
+    // initialValues can be object of strings, where entire form's shape is described
+    initialValues: PropTypes.object.isRequired,
+
+    onFinalStepSuccess: PropTypes.func.isRequired,
+    startingStepNumber: PropTypes.number,
+    steps: PropTypes.arrayOf(
+      PropTypes.shape({
+        stepRender: PropTypes.func.isRequired,
+        stepSubmit: PropTypes.func,
+        validationSchema: PropTypes.object.isRequired, // specifically a Yup object shape
+      }),
+    ).isRequired,
+  };
+
+  static defaultProps = {
+    startingStepNumber: 0,
+  };
+
+  state = {
+    // eslint-disable-next-line react/destructuring-assignment
+    stepNumber: this.props.startingStepNumber,
+    errorMessage: '',
+  };
+
+  showNextStep = () => {
+    this.setState(previousState => ({
```

Comment:
maybe do conditional logic here based upon the step's manifest? `shouldShow` maybe?